src/libc/pthread/mutex.rs

```diff
@@ -6,20 +6,28 @@
 //! Mutexes.
 
 use crate::dyld::{export_c_func, FunctionExports};
-use crate::libc::errno::{EDEADLK, EPERM};
+use crate::libc::errno::{EBUSY, EDEADLK, EPERM};
 use crate::mem::{ConstPtr, MutPtr, Ptr, SafeRead};
 use crate::{Environment, ThreadID};
 use std::collections::HashMap;
 use std::num::NonZeroU32;
 
 #[derive(Default)]
 pub struct State {
-    mutexes: HashMap<MutPtr<pthread_mutex_t>, MutexHostObject>,
+    // TODO?: Maybe this should be a Vec instead? It would be bad if there were many mutexes over
+    // the lifetime of an application, but it would perform better.
+    // Maybe it could also be a fixed size allocator? (although that seems a little overkill)
+    mutexes: HashMap<HostMutexId, MutexHostObject>,
```

Comment:
I think it'd probably be better to keep the mutex state struct separate. I'll move the struct down into `Environment` but keep it as is, unless you'd really prefer it the other way.

Reply:
To make sure I understand: you want to move this state struct  into `Environment`, rather than having it be in the pthread state struct? And by “keep it as is”, you mean the mutexes hashmap would still be inside the state struct, rather than being directly part of Environment? If so, that sounds good to me.

Reply:
I'm not exactly sure it's what you're thinking, but I have the fixes in. If it's not what you're thinking of, I'm open to changing it.

Reply:
c0a7a58551dc559fb5d45cb36bde886d821251f4 does more or less what I wanted, thanks!